src/server.c

```diff
@@ -3783,6 +3771,44 @@ int processCommand(client *c) {
         return C_OK;
     }
 
+    *valid = 1;
+    return C_OK;
+}
+
+/* If this function gets called we already read a whole
+ * command, arguments are in the client argv/argc fields.
+ * processCommand() execute the command or prepare the
+ * server for a bulk read from the client.
+ *
+ * If C_OK is returned the client is still alive and valid and
+ * other operations can be performed by the caller. Otherwise
+ * if C_ERR is returned the client was destroyed (i.e. after QUIT). */
+int processCommand(client *c) {
+
+    if (!(c->flags & CLIENT_RERUN_COMMAND)) {
+        int status, valid;
+        status = validateCommand(c, &valid);
+        if (!valid) return status;
+    }
```

Comment:
i think this change steps on the toes of #11012 and must be rebased and re-evaluated after.
see the now existing `client_reprocessing_command`.
@ranshid FYI

Reply:
Made related comment elsewhere. I'm not convinced we need this blocking code, but if we do we should try to align the various blocking approaches more.

Reply:
Madelyn, I think you are referring to the new *CLIENT* flag as opposed to the blocking code. I have removed this client flag but I think we will need a new blocking code that allows us to express the new contract of *resume the pending command after the command is replicated*.